src/core/build.gradle.kts

```diff
@@ -45,6 +46,9 @@ dependencies {
     api("org.apache.logging.log4j:log4j-core") {
         because("GuiLogEventAppender is using log4j-core to implement GUI-based log appender")
     }
+    kapt("org.apache.logging.log4j:log4j-core") {
```

Comment:
It is a bit weird that the annotation processor is located in the same artifact as `log4j-core` though. I would have expected something like `log4j-plugin-processor` artifact name for the processor.

Reply:
In Log4j 2.x a lot of classes were added to a single `log4j-core` artifact, since many users at that time didn't use dependency management.

In version 3.x there is a [`log4j-plugin-processor`](https://mvnrepository.com/artifact/org.apache.logging.log4j/log4j-plugin-processor) artifact.

Reply:
One more thing: the processor logs a couple of messages which I think do not belong to the default output.

```
Note: Processing Log4j annotations
Note: Annotations processed
Note: Processing Log4j annotations
Note: No elements to process
```

There's no value in those messages in the default build execution.